src/bundle/Controller/SectionController.php

```diff
@@ -490,9 +507,25 @@ private function getSectionsNumbers(array $sections): array
      * @param \eZ\Publish\API\Repository\Values\Content\Section $section
      *
      * @return bool
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
      */
     private function canUserAssignSectionToAnyContent(Section $section): bool
     {
-        return $this->isGranted(new Attribute('section', 'assign', ['valueObject' => new ContentInfo(), 'targets' => $section]));
+        $hasAccess = $this->permissionResolver->hasAccess('section', 'assign');
+
+        if (is_bool($hasAccess)) {
+            return $hasAccess;
+        }
+
+        foreach ($this->permissionUtil->flattenArrayOfLimitations($hasAccess) as $limitation) {
+            if ($limitation instanceof NewSectionLimitation) {
+                // If one of user limitation is NewSectionLimitation we check if user can assign Content to $section
+                return in_array($section->id, $limitation->limitationValues);
```

Comment:
As this is iterating several roles and policies, this should not return here unless value is true.
Afaik you can do like in `checkAllowedContentTypes()` and collect all allowed sections before you do a in_array.